core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -107,7 +107,9 @@ object Automation {
 
     fun tryTrainMilitaryUnit(city: City) {
         if (city.isPuppet) return
-        val chosenUnitName = chooseMilitaryUnit(city)
+        if (city.civInfo.gameInfo.ruleSet.units[city.cityConstructions.currentConstructionFromQueue]?.isMilitary() == true)
```

Comment:
There's a convenience function in cityConstructions to get the construction

Reply:
Done.